lib/net.js

```diff
@@ -527,6 +549,38 @@ Socket.prototype.end = function(data, encoding, callback) {
 };
 
 
+Socket.prototype.pause = function() {
+  if (this[kBuffer] && !this.connecting && this._handle &&
+      this._handle.reading) {
+    this._handle.reading = false;
+    if (!this.destroyed) {
+      const err = this._handle.readStop();
+      if (err)
+        this.destroy(errnoException(err, 'read'));
+    }
+  }
+  return stream.Duplex.prototype.pause.call(this);
+};
+
+
+Socket.prototype.resume = function() {
+  if (this[kBuffer] && !this.connecting && this._handle &&
+      !this._handle.reading) {
+    tryReadStart(this);
+  }
+  return stream.Duplex.prototype.resume.call(this);
+};
+
+
+Socket.prototype.read = function(n) {
+  if (this[kBuffer] && !this.connecting && this._handle &&
+      !this._handle.reading) {
+    tryReadStart(this);
+  }
+  return stream.Duplex.prototype.read.call(this, n);
+};
```

Comment:
Do you see another implementation that does not override these three methods?

Reply:
I'm not sure what you're really asking/getting at?

Reply:
I think overriding `read`,  `resume`, `pause`, `on`, etc.. is the source of most of the issues with streams.
I would prefer an implementation that did not do this all the time, but rather did this only if  `onread`  is specified, or even better not at all.

Reply:
@mcollina I don't see anything wrong with this approach. Can you elaborate?